### PR TITLE
Automated cherry pick of #96051: Allow priority to be set for kubelet process on Windows hosts

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
Cherry pick of #96051 on release-1.18.

#96051: Allow priority to be set for kubelet process on

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Introduce `windows-priorityclass` flag to set priority for the kubelet running on Windows nodes

**Which issue(s) this PR fixes**:

Automated cherry pick of #96051 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
Add a new flag to set priority for the kubelet on Windows nodes so that workloads cannot overwhelm the node there by disrupting kubelet process.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
Allow user to specify priority of the kubelet process on Windows nodes.
```

/sig windows
cc @marosset @jsturtevant @derekwaynecarr

### CODE REVIEW

**Comment by marosset** [comment]
Looks like the bazel builds need to be regenerated for this CP.

File: cmd/kubelet/app/init_windows.go [hunk 1]
```diff
@@ -19,14 +19,47 @@ limitations under the License.
...
+	return priorityClassMap[priorityClassName]
+}
+
+func initForOS(windowsService bool, windowsPriorityClass string) error {
+	priority := getPriorityValue(windowsPriorityClass)
+	if priority == 0 {
+		return fmt.Errorf("unknown priority class %s, valid ones are available at "+
+			"https://docs.microsoft.com/en-us/windows/win32/procthread/scheduling-priorities", windowsPriorityClass)
+	}
+	kubeletProcessHandle, err := windows.GetCurrentProcess()
```
Comment (ravisantoshgudimetla) [inline cmd/kubelet/app/init_windows.go:53]: I had to use `GetCurrentProcess()` as `CurrentProcess()` is not available

**Comment by derekwaynecarr** [comment]
why are we backporting features?

**Comment by derekwaynecarr** [comment]
see https://github.com/kubernetes/kubernetes/pull/96157#issuecomment-728253906

/approve
/lgtm

**Comment by ravisantoshgudimetla** [comment]
@kubernetes/release-managers for cherrypick approvals

**Comment by ravisantoshgudimetla** [comment]
@kubernetes/release-managers for cherrypick approvals

**Comment by cpanato** [comment]
/triage accepted

from parent PR
/priority important-soon

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by cpanato** [comment]
/triage accepted

from parent PR
/priority important-soon
/kind feature

**Pull request merged**